packages/embed/src/overlay/overlay.ts

```diff
@@ -47,20 +40,15 @@ export const createOverlay = (element: HTMLDivElement): Overlay => {
 
   element.appendChild(Container)
 
-  return {
-    hide: () => {
-      element.className = 'gr4vy__overlay gr4vy__overlay--hidden'
-    },
-    show: () => {
-      element.className = 'gr4vy__overlay'
-    },
+  const hide = () => {
+    element.className = 'gr4vy__overlay gr4vy__overlay--hidden'
+  }
+  const show = () => {
+    element.className = 'gr4vy__overlay'
   }
-}
 
-export const createOverlayController = (element: Overlay) => {
-  require('./overlay.css')
-  approvalStarted$.subscribe(element.show)
-  approvalCompleted$.subscribe(element.hide)
-  approvalCancelled$.subscribe(element.hide)
-  transactionFailed$.subscribe(element.hide)
+  subject.approvalStarted$.subscribe(show)
```

Comment:
We don't need to expose hide/show as its entirely from messages.
It could be worth looking at an `Actor` pattern.